resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/SparkKubernetesClientFactory.scala

```diff
@@ -128,17 +128,17 @@ object SparkKubernetesClientFactory extends Logging {
 
   object ClientType extends Enumeration {
     import scala.language.implicitConversions
-    val Driver = Val(DRIVER_CLIENT_REQUEST_TIMEOUT, DRIVER_CLIENT_CONNECTION_TIMEOUT)
-    val Submission = Val(SUBMISSION_CLIENT_REQUEST_TIMEOUT, SUBMISSION_CLIENT_CONNECTION_TIMEOUT)
+    val Driver = ClientTypeVal(DRIVER_CLIENT_REQUEST_TIMEOUT, DRIVER_CLIENT_CONNECTION_TIMEOUT)
```

Comment:
These enumerations should probably be put in a `scala-2` source along with adding a `scala-3` source with real enums that are compatible with Java. Since they are inline it might mean some code duplication.

Example: https://github.com/ekrich/sconfig/blob/main/sconfig/shared/src/main/scala-3/org/ekrich/config/ConfigValueType.scala

Reply:
This PR does not cross compile with Scala 3. This would come into a separate PR.
This is a small PR to make Scala 2.13 cross compilable. The big PR actually cross compiling would build on this one.